File: webapi/Models/Page.cs
namespace webapi.Models
{
    public record PageInfo
    {
        public required int PageNumber { get; set; }
        public required int PageSize { get; set; }
    }

    public record PageFilterInfo
    {
        public required int PageNumber { get; set; }
        public required int PageSize { get; set; }
        public required Filter Filter { get; set; }

    }

    public record struct PageResponse<T> (
        int Pages, int CountRecords, List<T> Records
    );

}

File: reactapp/src/models/Page/Page.ts
import { defaultFilter, Filter } from "../Filter"

export interface PageInfo {
    pageNumber: int,
    pageSize: int,
}


export const initialPageState: PageInfo = {
    pageNumber: 1,
    pageSize: 100
}


export interface PageFilterInfo {
    pageNumber: int,
    pageSize: int,
    filter: Filter
}


export const defautlPageFilterInfo:PageFilterInfo = {
    pageNumber: 0,
    pageSize: 0,
    filter: defaultFilter
}


export function createPageFilterInfo(
    pageNumber: int, pageSize: int, filter: Filter
): PageFilterInfo {
    return {
        pageNumber,
        pageSize,
        filter
    }
} 